app/Http/Request/AdsRequest.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace App\Http\Request;
+
+use Anik\Form\FormRequest;
+use Illuminate\Http\JsonResponse;
+
+class AdsRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    protected function authorize(): bool
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    protected function rules(): array
+    {
+        if ($this->method() == 'POST') {
+            return [
+                'page_name' => 'required|string'
+            ];
+        }
+    }
+
+    /**
+     * format the validation response if there is error in validation.
+     *
+     * @return array
+     */
+    protected function apiJsonResponse(): ?apiJsonResponse
+    {
+
+        return response()->json([
```

Comment:
Can we use this?
`return $this->resProvider->apiJsonResponse(422, $this->errorMessage(), '', $this->validator->errors()->messages());`